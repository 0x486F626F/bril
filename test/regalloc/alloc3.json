{
  "functions": [
    {
      "instrs": [
        {
          "dest": "vj",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "dest": "vk",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "dest": "v1",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "dest": "v2",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "dest": "v3",
          "op": "const",
          "type": "int",
          "value": 3
        },
        {
          "dest": "v0",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "args": [
            "vj",
            "v1"
          ],
          "dest": "vg",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vk",
            "v0"
          ],
          "dest": "vh",
          "op": "sub",
          "type": "int"
        },
        {
          "args": [
            "vg",
            "vh"
          ],
          "dest": "vf",
          "op": "mul",
          "type": "int"
        },
        {
          "args": [
            "vj",
            "v2"
          ],
          "dest": "ve",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vj",
            "v3"
          ],
          "dest": "vm",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vf"
          ],
          "dest": "vb",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "ve",
            "v0"
          ],
          "dest": "vc",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vc"
          ],
          "dest": "vd",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "vm",
            "v0"
          ],
          "dest": "vk",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vb"
          ],
          "dest": "vj",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "vj"
          ],
          "op": "print"
        },
        {
          "args": [
            "vk"
          ],
          "op": "print"
        },
        {
          "args": [
            "vd"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
