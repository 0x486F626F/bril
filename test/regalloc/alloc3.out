main {
	vj: int = const 1;
	__r03: int = const 2;
	__r01: int = const 1;
	__r04: int = const 2;
	v3: int = const 3;
	v0: int = const 5;
	__r02: int = add vj __r01;
	v1: int = id __r01;
	__r01: int = sub __r03 v0;
	vf: int = id __r02;
	__r02: int = id vg;
	__r02: int = mul __r02 __r01;
	vh: int = id __r01;
	__r01: int = add vj __r04;
	vk: int = id __r03;
	__r03: int = add vj v3;
	vb: int = id __r02;
	__r02: int = id vf;
	__r02: int = id __r02;
	ve: int = id __r01;
	vc: int = id __r01;
	__r01: int = id ve;
	__r01: int = add __r01 v0;
	vd: int = id __r01;
	__r01: int = id vc;
	__r01: int = id __r01;
	vm: int = id __r03;
	vk: int = id __r03;
	__r03: int = id vm;
	__r03: int = add __r03 v0;
	vj: int = id vb;
	print vj;
	print __r03;
	print __r01;
}
